RevenueCatUI/CustomerCenter/ViewModels/BaseManageSubscriptionViewModel/BaseManageSubscriptionViewModel.swift

```diff
@@ -0,0 +1,337 @@
+                self.inAppBrowserURL = IdentifiableURL(url: url)
+            } else {
+                // WIP: shor error alert?
+            }
+
+        case .changePlans, .cancel:
+            self.actionWrapper.handleAction(.showingManageSubscriptions)
+
+        case .customUrl:
+            guard let url = path.url,
+                  let openMethod = path.openMethod else {
+                Logger.warning("Found a custom URL path without a URL or open method. Ignoring tap.")
+                return
+            }
+            switch openMethod {
+            case .external,
+                _ where !url.isWebLink:
+                URLUtilities.openURLIfNotAppExtension(url)
+            case .inApp:
+                self.inAppBrowserURL = .init(url: url)
+            @unknown default:
+                Logger.warning(Strings.could_not_determine_type_of_custom_url)
+                URLUtilities.openURLIfNotAppExtension(url)
+            }
+
+        default:
+            break
+        }
+    }
+#endif
+
+}
+
+private extension CustomerCenterConfigData.Screen {
+
+    var filteredPaths: [CustomerCenterConfigData.HelpPath] {
+        return self.paths.filter { path in
+            return path.type != .unknown
+        }
+    }
+
+}
+
+private extension Array<CustomerCenterConfigData.HelpPath> {
+    func relevantPathsForPurchase(
+        _ purchaseInformation: PurchaseInformation?
+    ) -> [CustomerCenterConfigData.HelpPath] {
+        guard let purchaseInformation else {
+            return self
+        }
+
+        return filter {
+            let isNonAppStorePurchase = purchaseInformation.store != .appStore
+            let isAppStoreOnlyPath = $0.type.isAppStoreOnly
+
+            // skip AppStore only paths if the purchase is not from App Store
+            guard !(isNonAppStorePurchase && isAppStoreOnlyPath) else {
+                return false
+            }
+
+            // if it's cancel, it cannot be a lifetime subscription
+            let isCancel = $0.type == .cancel
+            let isEligibleCancel = !purchaseInformation.isLifetime && !purchaseInformation.isCancelled
+
+            // if it's refundRequest, it cannot be free  nor within trial period
+            let isRefund = $0.type == .refundRequest
+            let isRefundEligible = purchaseInformation.price != .free
+                                    && !purchaseInformation.isTrial
+                                    && !purchaseInformation.isCancelled
+
+            // if it has a refundDuration, check it's still valid
+            let refundWindowIsValid = $0.refundWindowDuration?.isWithin(purchaseInformation) ?? true
+
+            return (!isCancel || isEligibleCancel) &&
+                    (!isRefund || isRefundEligible) &&
+                    refundWindowIsValid
+        }
+    }
+}
+
+private extension CustomerCenterConfigData.HelpPath.PathType {
+
+    var isAppStoreOnly: Bool {
+        switch self {
+        case .cancel, .customUrl, .missingPurchase:
```

Comment:
`.missingPurchase` should also be appStore only because it's not possible to restore the purchases